### ISSUE TITLE
[BUG] Unexpected Generator From a Test Result - Request for Docs Update? (SDK vs CLI)

### ISSUE THREAD

**Issue by mindlessroman** [microsoft/promptflow#590]
**Describe the bug**
Using a node that uses the SDK for the AOAI tool returns what I expect, similar to how the examples/tutorials show. Using the package/tool in the flow.dag.yaml returns the <generator object generate_from_proxy at MEMORY_LOCATION>. 

**How To Reproduce the bug**
Steps to reproduce the behavior, how frequent can you experience the bug:
1. Try to mimic what's shown in this example [Basic Flow with Builtin LLM](https://github.com/microsoft/promptflow/blob/main/examples/flows/standard/basic-with-builtin-llm/flow.dag.yaml):
    ```yaml
    inputs:
      question:
        type: string
    outputs:
      answer:
        type: string
        reference: ${answer_the_question_with_context.output}
    nodes:
      ...
      - name: answer_the_question_with_context
        type: llm
        source:
          type: code
          path: step05_answer_question_with_context.jinja2
        inputs:
          deployment_name: gpt-35-turbo
          temperature: 0
          max_tokens: 1000
          prompt_text: ${prompt_variants.output}
        connection: some-aoai-connection
        api: chat
    ```
2. Test the flow in a jupyter notebook:
    ```python
    test_result = pf.test(flow="some-flow-name", inputs={"question":"What are bugs?")
    print(f"Flow outputs: {test_result}")
    ```
    ```text
    Flow outputs: {'answer': <generator object generate_from_proxy at 0x177971740>}
    ```
3. Test the flow using the CLI:
    ```shell
    $ pf flow test --flow some-flow-name --inputs question="What are bugs?"
    ```
    ```text
    {
        "answer": "Blah blah blah I'm a relevant answer"
    }
    ```

**Expected behavior**

I'd suspect that both would return the same thing, but it seems that when printed -- which many of the example notebooks do with the test result -- there's no indication that some formatting will be required.

Workaround ended up being:
```python
test_result = [i for i in pf.test(flow=FLOW_NAME, inputs=inputs)["answer"]]
print(f"Flow outputs: {''.join(test_result)}")
```

If one doesn't use the built in LLM package and instead use the SDK in a python code node, this error doesn't come up.

```yaml
inputs:
  question:
    type: string
outputs:
  answer:
    type: string
    reference: ${answer_the_question_with_context.output}
nodes:
...
- name: answer_the_question_with_context
  type: python
  source:
    type: code
    path: step05_answer_question_with_context.py
  inputs:
    prompt_text: ${STEP_FOUR.output}
    aoai_connection: some-aoai-connection
```
`step05_answer_question_with_context.py`: Connects to AOAI
```python
from promptflow import tool
from promptflow.connections import AzureOpenAIConnection
from promptflow.tools import AzureOpenAI
import os
from jinja2 import Template as jinja_template

@tool
def answer_question_with_context(
    prompt_text: jinja_template, 
    aoai_connection: AzureOpenAIConnection
) -> str:

    aoai_gpt_deployment = os.getenv("AZURE_OPENAI_GPT_DEPLOYMENT")

    response = AzureOpenAI(connection=aoai_connection).chat(
        prompt_text, 
        deployment_name=aoai_gpt_deployment, 
        temperature=0, 
        max_tokens=1000
    )

    return response
```

I think this all comes from being bleeding/cutting edge.

**Screenshots**

**Running Information(please complete the following information):**
 - Promptflow Package Version using `pf -v`: 0.1.0b6
 - Operating System: Mac OS 13.5.2
 - Python Version using `python --version`: Python 3.9.17

**Additional context**
Branching from #561.

**Comment by lalala123123** [comment]
Hi @mindlessroman , this is a bug in the Prompt Flow SDK and should ensure that the user experience of standard flow is consistent across the CLI and SDK. It will be fixed in #660

**Comment by lalala123123** [comment]
It has been fixed in the latest version, please try to upgrade to 0.1.0b8
